test/index-spec.js

```diff
@@ -205,18 +205,23 @@ describe('with valid inputs', () => {
     },
   }];
 
-  testCases.forEach(testCase => it(testCase.description, async () => {
+  const testBody = (testCase, withGetSignedUrl) => async () => {
     const validateRequest = sinon.stub().returns(true);
-    const S3 = {
-      getSignedUrl: (command, params, callback) => {
-        callback(null, 'signed-url');
-      }
-    };
 
     const middleware = factory(Object.assign({
       bucket: 'bucket',
       validateRequest,
-      S3,
+      ...(withGetSignedUrl ? {
+        getSignedUrl: (/* command, params */) => {
```

Comment:
do we need commented out text?

Reply:
not really, but i just wanted to indicate what params it uses, which we don't need for mocked function